### PR TITLE
Php code cleanup

### PR DESCRIPTION
**Pull request by greylabel**
First of a couple PRs related to cleanup/modernizing before bumping to twig 3, and then update all the JS packages and implementation.

- Start of adding types to PHP code
- Various code styles cleanups and renaming
- Make use of PHP 8.2 features

### CODE REVIEW

File: __tests__/twig-extensions/alter-twig.php [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <?php
 
+// declare(strict_types=1);
```
Comment (greylabel) [inline __tests__/twig-extensions/alter-twig.php:3]: Will enable this for all files in an upcoming PR.

File: __tests__/twig-extensions/alter-twig.php [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+
 use Twig\Environment;
 use Twig\TwigFunction;
 
-function addCustomExtension(Environment &$env, $config) {
-  $env->addFunction(new TwigFunction('customTwigFunctionThatSaysWorld', fn() => 'Custom World'));
+function addCustomExtension(Environment &$twigEnvironment, $config): void {
```
Comment (greylabel) [inline __tests__/twig-extensions/alter-twig.php:8]: Changes in this PR make use of new arrow function syntax and typing.

File: src/TwigRenderer.php [hunk 3]
```diff
@@ -11,25 +13,19 @@ class TwigRenderer {
...
-
-  /**
-   * @var $config array - Configuration passed in
-   */
-  public $config;
+  private readonly \Twig\Loader\ChainLoader $loaders;
 
-  function __construct(array $config) {
-    $this->config = $config;
+  function __construct(public array $config) {
```
Comment (greylabel) [inline src/TwigRenderer.php:28]: `$config` makes use of new constructor promotion feature.

File: src/server--async.php [hunk 4]
```diff
@@ -34,7 +36,7 @@
 }
 
 try {
-  $config = json_decode($configString, true);
+  $config = json_decode($configString, true, 512, JSON_THROW_ON_ERROR);
```
Comment (greylabel) [inline src/server--async.php:39]: New `JSON_THROW_ON_ERROR` is a better error experience with `json_decode` and `json_encode`

**Review by EvanLovely — APPROVED** [review]

**Pull request merged**